apps/frontend/src/components/filters.tsx

```diff
@@ -0,0 +1,250 @@
+  const [latitude, setLatitude] = useState<number>(48.463302);
+  const [longitude, setLongitude] = useState<number>(-123.3108);
+
+  const handleMinPriceChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setMinPrice(
+      event.target.value === "" ? null : parseFloat(event.target.value),
+    );
+  };
+
+  const handleMaxPriceChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setMaxPrice(
+      event.target.value === "" ? null : parseFloat(event.target.value),
+    );
+  };
+
+  const handleStatusChange = (event: SelectChangeEvent<string>) => {
+    setStatus(event.target.value);
+  };
+
+  const handleTypeChange = (event: SelectChangeEvent<string>) => {
+    setType(event.target.value);
+  };
+
+  const handleApplyFilters = () => {
+    // Pass the current filter values to the parent component only when "Apply Filters" button is clicked
+    onFilterChange({
+      minPrice: minPrice,
+      maxPrice: maxPrice,
+      status: status,
+      searchType: type,
+      latitude: latitude,
+      longitude: longitude,
+    });
+  };
+
+  const handleClearFilters = () => {
+    setMinPrice(null);
+    setMaxPrice(null);
+    setStatus("");
+    setType("LISTING");
+    onFilterChange({
+      minPrice: null,
+      maxPrice: null,
+      status: "",
+      searchType: "",
+      latitude: 48.463302,
+      longitude: -123.3108,
+    });
+  };
+
+  // Gets the user location, returns false on failure, defaults to uvic
+  const getLocation = () => {
+    try {
+      navigator.geolocation.getCurrentPosition((position) => {
+        const currentLatitude = position.coords.latitude;
+        const currentLongitude = position.coords.longitude;
+        if (currentLatitude !== 0 && currentLongitude !== 0) {
+          setLatitude(currentLatitude);
+          setLongitude(currentLongitude);
+          handleApplyFilters();
+          return true;
+        }
+      });
+      return false;
+    } catch (error) {
+      return false;
+    }
+  };
+
+  useEffect(() => {
+    setMinPrice(filters.minPrice);
+    setMaxPrice(filters.maxPrice);
+    setStatus(filters.status);
+    setType(filters.searchType);
+    setLatitude(filters.latitude);
+    setLongitude(filters.longitude);
+  }, [filters]);
+
+  const isDesktop = useMediaQuery("(min-width:850px)");
+
+  return (
+    <Grid item container xs={isDesktop ? 5 : 10}>
+      <Typography variant="h6" component="h3" m={1}>
+        Filters
+      </Typography>
+      <Box
+        sx={{
+          display: "flex",
+          flexDirection: "row",
+          alignItems: "center",
+          justifyContent: "center",
+          width: "100%",
+        }}
+      >
+        <FormControl fullWidth sx={{ m: 1 }}>
+          <FormHelperText>Min</FormHelperText>
+          <OutlinedInput
+            id="outlined-adornment-amount"
+            startAdornment={<InputAdornment position="start">$</InputAdornment>}
+            placeholder="Min"
+            value={minPrice !== null ? minPrice : ""}
```

Comment:
Since we're being marked on form validation, we need to make sure that these values are handled if they aren't valid numbers. Luckily this should be pretty easy to add, for example, my price field on the CreateListing. 

```tsx
<TextField
  id="field-price"
  label="Price(CAD)"
  type="number"
  sx={{ m: "10px" }}
  rows={1}
  InputProps={{ inputProps: { min: 0 } }}
  onChange={updateListingPrice}
  error={!!priceError}
/>
{priceError && (
  <FormHelperText error>{priceError}</FormHelperText>
)}
```

The above works by updating the price error(a useState string type variable) on change if the price value entered doesn't match the regex of `const regex = /^\d+(.\d{1,2})?$/;` which should be every positive price such as:

- 10
- 1
- 21.1
- 12.11

Here is my updateListingPrice function for your reference

```tsx
const updateListingPrice = (event: ChangeEvent<HTMLInputElement>) => {
    // Handle
    const regex = /^\d+(.\d{1,2})?$/;
    if (!regex.test(event.target.value)) {
      setPriceError(
        "This price is not valid, please make sure the value is positive and in the form xx.xx",
      );
    } else {
      setPriceError("");
      const priceValue: number = +event.target.value;
      updateNewListingPayload(
        "price",
        priceValue >= 0 ? priceValue : priceValue * -1,
      );
    }
  };
```

I believe implementing this will also fix the issue where if the value entered into the price fields is not a number it doesn't get cleared by the "Clear Filter" button 

![image](https://github.com/UVicMartletplace/martletplace/assets/54737724/348b9a8f-9efe-4d4a-994c-e93063611ee1)